app/models/user.rb

```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  has_many :user_trails
+  has_many :trails, through: :user_trails
+
+  validates_presence_of :name
+  validates_presence_of :email
+
+  def self.find_by_user_id(id)
```

Comment:
Since Rails already has a built in `find_by` method where you can pass id (`find` also defaults to finding by ID, but it errors out differently if it doesn't find anything), it's generally  best not to duplicate such a similar method. Can you refactor your controller to just use one of the built in find methods so that you don't need this additional custom find method?